source/jormungandr/tests/line_sections_tests.py

```diff
@@ -410,32 +703,80 @@ def test_stop_schedule_impacted_by_line_section(self):
+            assert result == (disruption in d)
+        assert not impacted_ids(d)
 
-        r = self.query_region('stop_areas/C/stop_schedules?{cur}&{d}&{f}'.format(cur=cur, d=dt, f=fresh))
+        # B
+        scenario = {
+            'line_section_on_line_1': False,
+            'line_section_on_line_1_other_effect': False,
+            'line_section_on_line_2': True,
+        }
+        r = self.query_region(query.format(sa='B', cur=cur, d=dt, f=fresh))
         d = get_all_element_disruptions(r['stop_schedules'], r)
-        assert 'line_section_on_line_1' in d
-        assert impacted_ids(d) == {
-            'vj:1:1'
-        }  # the impact is linked in the response only to the stop_schedule.datetime
+        for disruption, result in scenario.iteritems():
+            assert result == (disruption in d)
+        assert impacted_ids(d) == {'vj:2'}
 
-        r = self.query_region('stop_areas/D/stop_schedules?{cur}&{d}&{f}'.format(cur=cur, d=dt, f=fresh))
+        # C
+        scenario = {
+            'line_section_on_line_1': True,
+            'line_section_on_line_1_other_effect': False,
+            'line_section_on_line_2': False,
+        }
+        r = self.query_region(query.format(sa='C', cur=cur, d=dt, f=fresh))
         d = get_all_element_disruptions(r['stop_schedules'], r)
-        assert 'line_section_on_line_1' in d
+        for disruption, result in scenario.iteritems():
+            assert result == (disruption in d)
+        # the impact is linked in the response only to the stop_schedule.datetime
         assert impacted_ids(d) == {'vj:1:1'}
 
-        r = self.query_region('stop_areas/E/stop_schedules?{cur}&{d}&{f}'.format(cur=cur, d=dt, f=fresh))
+        # D
+        scenario = {
+            'line_section_on_line_1': True,
+            'line_section_on_line_1_other_effect': False,
+            'line_section_on_line_2': False,
+        }
+        r = self.query_region(query.format(sa='D', cur=cur, d=dt, f=fresh))
         d = get_all_element_disruptions(r['stop_schedules'], r)
-        assert 'line_section_on_line_1' in d
+        for disruption, result in scenario.iteritems():
+            assert result == (disruption in d)
         assert impacted_ids(d) == {'vj:1:1'}
 
-        r = self.query_region('stop_areas/F/stop_schedules?{cur}&{d}&{f}'.format(cur=cur, d=dt, f=fresh))
+        # E
+        scenario = {
+            'line_section_on_line_1': True,
+            'line_section_on_line_1_other_effect': True,
+            'line_section_on_line_2': False,
+        }
+        r = self.query_region(query.format(sa='E', cur=cur, d=dt, f=fresh))
+        d = get_all_element_disruptions(r['stop_schedules'], r)
+        for disruption, result in scenario.iteritems():
+            assert result == (disruption in d)
+        assert impacted_ids(d) == {'vj:1:1'}
+
+        # F
+        scenario = {
+            'line_section_on_line_1': False,
+            'line_section_on_line_1_other_effect': False,
```

Comment:
OK, so it's False for `line_section_on_line_1_other_effect` and `line_section_on_line_2` just because stop_schedule is empty in concerned directions, here I guess.